src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs

```diff
@@ -189,10 +189,16 @@ protected void ReadSubIfd(List<IExifValue> values)
     {
         if (this.subIfds is not null)
         {
-            foreach (ulong subIfdOffset in this.subIfds)
+            do
             {
-                this.ReadValues(values, (uint)subIfdOffset);
+                ulong[] buf = [.. this.subIfds];
+                this.subIfds.Clear();
+                foreach (ulong subIfdOffset in buf)
+                {
+                    this.ReadValues(values, (uint)subIfdOffset);
+                }
             }
+            while (this.subIfds.Count > 0);
```

Comment:
L195 clears the list (so count = 0). Is this condition necessary then?

Reply:
L198 can add to `subIfds` (nested sub ifd(s))